app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt

```diff
@@ -0,0 +1,59 @@
+package space.taran.arkshelf.data
+
+import android.content.Context
+import com.beust.klaxon.Klaxon
+import space.taran.arkshelf.domain.Link
+import java.nio.file.Path
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+import kotlin.io.path.Path
+import kotlin.io.path.deleteIfExists
+import kotlin.io.path.inputStream
+import kotlin.io.path.name
+import kotlin.io.path.outputStream
+import kotlin.io.path.writeText
+
+class LinkLocalDataSource(private val context: Context) {
+    private val klaxon = Klaxon()
+    private val jsonTmpPath =
+        Path(context.cacheDir.absolutePath).resolve(Path("link.json"))
```

Comment:
`createTempFile()` from Kotlin standard library create files in cache dir too. But in this case, there will be a lot of files that will need to be cleaned. Right now there is only one file that is being reused.

Reply:
Still it's better to create random file than reuse a file with fixed path. If we will parallelize code at some point (maybe automatically grabbing links from some resource?) than this would be a point of synchronization.

> createTempFile() from Kotlin standard library create files in cache dir too
I'm surprized, I think it's a flaw.